### PR TITLE
feat: detect and fix nested link wrappers in upgrade wizard

### LINKED ISSUE #667: upgrade dev-main 1d34892 => dev-main 0d7dedb
Hi,

sorry to say that I found some bugs/inconsistencies in the brand new version.

**links in inline linked images:** Double-a-tags in BE and FE still present (and Standalone.html is used).

BE:
```
<p>
    <a class="image image-inline" href="t3://page?uid=1#1" target="_blank"><a class="image image-inline" href="t3://page?uid=1#1" target="_blank"><img class="image-inline" src="/fileadmin/user_upload/moto_dog_1.jpg" data-htmlarea-file-uid="2" data-htmlarea-file-table="sys_file" width="224" height="300" title="" alt="" data-title-override="true" data-alt-override="true" data-quality="retina"></a></a>
</p>
```

FE:
`<p><a class="image image-inline" href="#c1" target="_blank"></a><a class="image image-inline" href="#c1" target="_blank"></a></p>`

**German Umlauts and other non-ASCII characters:** Seems to be okay, even nbsp are preserved. 

```
<figure class="image image-left">
    <a class="image" href="t3://page?uid=1#1"><img src="/fileadmin/user_upload/moto_dog_1.jpg" data-htmlarea-file-uid="2" data-htmlarea-file-table="sys_file" width="75" height="100" title="" alt="" data-title-override="true" data-alt-override="true" data-quality="retina" data-caption="A very long figcaption for such a small image ÄÜÖ äüö ß €  &nbsp;&nbsp;&nbsp;∝¼¢₡₵"></a>
    <figcaption>
        A very long figcaption for such a small image ÄÜÖ äüö ß € &nbsp;&nbsp;&nbsp;&nbsp;∝¼¢₡₵
    </figcaption>
</figure>
```

Your hint is correct but without consequences: css-controlled wrapping don't work. Word-wrap has no effect on figcaption because the figcaption itself has no width specification (img-width has to be added to figcaption to make it work; {image.width}  in partial for FE).

Image

**Differences for figcatpiton in table and image:**
I'm a little bit surprised that figcaptions in tables preserve br (and other tags like a-tags) while img-figcaptions strips them. Maybe the cause is not ckeditor itself but the modifications for TYPO3 (eventually you can find something useful at https://forge.typo3.org/issues/101874).

If I add some style-definitions in my yaml, the class shows up in dropdown and is applied and stored for tables
    style:
      definitions:
        - { name: "Tbl figcaption", element: "figcaption", classes: ['tbl-figcaption-class'] }

For images the dropdown don't show the class and when adding something in code-view the tag is removed after toggling to normal-view. Even the buttons for strong, italic remove the so formated figcaption-text when the save-button is clicked. 

Image
FE:

Image

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Extends the `RteImageReferenceValidator` to detect and auto-fix `<a><a></a></a>` double wrapping in stored RTE content ([#667](https://github.com/netresearch/t3x-rte_ckeditor_image/issues/667)).

This complements [PR #669](https://github.com/netresearch/t3x-rte_ckeditor_image/pull/669) (CKEditor upcast + `renderInlineLink()` fix) by allowing editors to batch-fix affected content elements via the upgrade wizard instead of manually opening and saving each one.

### Changes

- **`ValidationIssueType`**: Add `NestedLinkWrapper` enum case
- **`RteImageReferenceValidator`**: 
  - `detectNestedLinkWrappers()` — regex-based detection of `<a><a>...</a></a>` patterns
  - `collapseNestedLinks()` — structural fix that strips inner `<a>` wrapper, keeping outer attributes. Loops to handle triple+ nesting
  - Both called from existing `validateHtml()` and `applyFixes()` pipelines
- **`RteImagePreviewRenderer`**: Add `NestedLinkWrapper` to exhaustive match expression
- **13 unit tests** covering detection (5), fixing (5), edge cases (3: mixed issue types, fixable issues integration, triple nesting)

### How it works

1. `validateHtml()` runs existing per-`` checks, then appends results from `detectNestedLinkWrappers()` 
2. `applyFixes()` calls `collapseNestedLinks()` first (structural fix), then proceeds with existing src-based fixes
3. The two fix types compose correctly — a nested-linked image with a wrong src gets both fixes applied

## Test plan

- [x] 700/700 unit tests pass (`composer ci:test:php:unit`)
- [x] PHPStan level 10+ clean (`composer ci:test:php:phpstan`)
- [x] Pre-commit hooks pass (phplint, php-cs-fixer, phpstan)
- [x] 3 internal code review cycles completed
- [ ] CI passes (all workflows)
- [ ] Verify with E2E tests

Closes #667

### CODE REVIEW

File: Classes/Service/RteImageReferenceValidator.php [hunk 1]
```diff
@@ -451,6 +457,78 @@ private function groupIssuesByRecord(array $issues): array
...
+
+        $pattern = '/(<a\b[^>]*>)\s*<a\b[^>]*>\s*(<img\b[^>]*\bdata-htmlarea-file-uid\b[^>]*>)\s*<\/a>\s*(<\/a>)/i';
+
+        // Loop to handle triple+ nesting: each pass removes one layer
+        $count = 0;
+
+        do {
+            $collapsed = preg_replace($pattern, '$1$2$3', $html, -1, $count);
+            $html      = is_string($collapsed) ? $collapsed : $html;
+        } while ($count > 0);
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Service/RteImageReferenceValidator.php]: Good catch on the ReDoS risk. Added an iteration limit of 10 to the do-while loop (commit 8bd3df8). Each pass removes exactly one nesting layer, so 10 iterations handles up to 11-deep nesting — far beyond any realistic scenario. The loop now terminates deterministically regardless of input.

File: Classes/Service/RteImageReferenceValidator.php [hunk 2]
```diff
@@ -451,6 +457,78 @@ private function groupIssuesByRecord(array $issues): array
...
+                break;
+            }
+        }
+
+        if (!$hasNestedLinkIssues) {
+            return $html;
+        }
+
+        $pattern = '/(<a\b[^>]*>)\s*<a\b[^>]*>\s*(<img\b[^>]*\bdata-htmlarea-file-uid\b[^>]*>)\s*<\/a>\s*(<\/a>)/i';
+
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Service/RteImageReferenceValidator.php]: Outer-wins is intentional and correct for this specific bug. In issue #667, `renderInlineLink()` adds the outer `<a>` with resolved attributes (t3:// → real URL), while the inner `<a>` retains the original unresolved attributes. In practice, both `<a>` tags have identical attributes since the bug duplicates them — so either-wins would produce the same result.

DOMDocument cannot be used here because it normalizes invalid nested `<a>` tags (auto-closes the outer when encountering the inner), making the nesting undetectable via DOM traversal. Regex is the correct tool for this specific pattern.

Added documentation explaining this rationale in the method docblock (commit 8bd3df8).
